src/librustc/ich/impls_syntax.rs

```diff
@@ -18,7 +18,12 @@ use rustc_data_structures::stable_hasher::{HashStable, StableHasher};
 
 impl<'ctx> rustc_target::StableHashingContextLike for StableHashingContext<'ctx> {}
 
-impl_stable_hash_for!(struct ::syntax::ast::Lifetime { id, ident });
+impl<'a> HashStable<StableHashingContext<'a>> for ast::Lifetime {
```

Comment:
This could use `HashStable_Generic` now?

Reply:
It requires hashing the `NodeId`. I have a commit with this change, I can add it to this PR.

Reply:
Ah, ok. You can leave that for a later PR.